cli/command/plugin/create_test.go

```diff
@@ -0,0 +1,150 @@
+package plugin
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/docker/cli/internal/test"
+	"github.com/docker/cli/internal/test/testutil"
+	"github.com/docker/docker/api/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateErrors(t *testing.T) {
+
+	testCases := []struct {
+		args          []string
+		expectedError string
+	}{
+		{
+			args:          []string{},
+			expectedError: "requires at least 2 arguments",
+		},
+		{
+			args:          []string{"INVALID_TAG", "context-dir"},
+			expectedError: "invalid",
+		},
+		{
+			args:          []string{"plugin-foo", "nonexistent_context_dir"},
+			expectedError: "no such file or directory",
+		},
+	}
+
+	for _, tc := range testCases {
+		cli := test.NewFakeCli(&fakeClient{})
+		cmd := newCreateCommand(cli)
+		cmd.SetArgs(tc.args)
+		cmd.SetOutput(ioutil.Discard)
+		testutil.ErrorContains(t, cmd.Execute(), tc.expectedError)
+	}
+}
+
+func TestCreateErrorOnFileAsContextDir(t *testing.T) {
+	tmpFile, err := ioutil.TempFile("", "file-as-context-dir")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(tmpFile.Name())
+
+	cli := test.NewFakeCli(&fakeClient{})
+	cmd := newCreateCommand(cli)
+	cmd.SetArgs([]string{"plugin-foo", tmpFile.Name()})
+	cmd.SetOutput(ioutil.Discard)
+	testutil.ErrorContains(t, cmd.Execute(), "context must be a directory")
+}
+
+func TestCreateErrorOnContextDirWithoutConfig(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "plugin-create-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	cli := test.NewFakeCli(&fakeClient{})
+	cmd := newCreateCommand(cli)
+	cmd.SetArgs([]string{"plugin-foo", tmpDir})
+	cmd.SetOutput(ioutil.Discard)
+	testutil.ErrorContains(t, cmd.Execute(), "config.json: no such file or directory")
+}
+
+func TestCreateErrorOnInvalidConfig(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "plugin-create-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	err = setupPluginDir(tmpDir, []byte("invalid-config-contents"))
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
This can be do with (`gotestyourself/fs`):

```go
tmpDir := fs.NewDir(t, "plugin-create-test",
    fs.WithDir("rootfs",
        fs.WithFile("config.json", "invalid-config-contents")))
defer tmpDir.Remove()
```

Just as an FYI, I think this works for now.

Reply:
Thank you for the review. The `gotestyourself` looks very interesting (just taking a look at the above example, it reads much better than the function I wrote for these tests). 

If you are OK with such a change, I can change to use `gotestyourself/fs` here.

Reply:
That would be great, thanks!